avocado/utils/crypto.py

```diff
@@ -50,14 +57,27 @@ def hash_file(filename, size=None, algorithm="md5"):
         return None
 
     with open(filename, "rb") as file_to_hash:
-        while size > 0:
-            if chunksize > size:
-                chunksize = size
-            data = file_to_hash.read(chunksize)
-            if len(data) == 0:
-                LOG.debug("Nothing left to read but size=%d", size)
-                break
-            hash_obj.update(data)
-            size -= len(data)
+        if num_threads > 1:
```

Comment:
Correct me if I am wrong, but you should be able to use `ThreadPoolExecutor` even then the `num_threads=1`. Therefore, the `if` is not necessary and you should be able to remove the code repetition.